packages/core/data-transfer/src/engine/index.ts

```diff
@@ -364,15 +364,19 @@ class TransferEngine<
               const path = diff.path.join('.');
 
               if (diff.kind === 'added') {
-                return `Added "${path}": "${diff.value}" (${diff.type})`;
+                return `${path} exists in destination schema but not in source schema`;
               }
 
               if (diff.kind === 'deleted') {
-                return `Removed "${path}"`;
+                return `${path} exists in source schema but not in destination schema`;
               }
 
               if (diff.kind === 'modified') {
-                return `Modified "${path}": "${diff.values[0]}" (${diff.types[0]}) => "${diff.values[1]}" (${diff.types[1]})`;
+                return `Schema value changed at "${path}": "${diff.values[0]}" (${diff.types[0]}) => "${diff.values[1]}" (${diff.types[1]})`;
+              }
+
+              if (diff.kind === 'dataType') {
```

Comment:
Are we sure we need this? Technically it's a subset scenario of modified, isn't it?
Since typeof values are returned in the modified payload, checking for different data types would be diff.types[0] !== diff.types[1].

Reply:
Yeah, true. I thought it would be nice to have a simplified way to check those cases, mostly because I originally thought I was solving a bug that involved it. But I think I may just reduce this PR to only the messaging changes and we can worry about these cases later once it's necessary.